### PR TITLE
Validate batch cleanup

### PR DESCRIPTION
**Pull request by robert-seitz-uniswap**
Description:

add batch validate calls instead of calling validate for individual transactions.
optimize other onchain reads for block number/fillevents
add metrics for dynamo reads, batchvalidate, onchain calls 
parallelize (batched) calls in main loop

### CODE REVIEW

File: bin/stacks/status-stack.ts [hunk 1]
```diff
@@ -67,6 +67,9 @@ export class StatusStack extends cdk.NestedStack {
       taskDefinition,
       desiredCount: 1,
       healthCheckGracePeriod: Duration.seconds(60),
+      circuitBreaker: {
+        rollback: true,
+      },
```
Comment (robert-seitz-uniswap) [inline bin/stacks/status-stack.ts:72]: adds rollback on task deploy failures

File: lib/compute/on-chain-status-checker.ts [hunk 2]
```diff
@@ -22,44 +27,34 @@ export class OnChainStatusChecker {
...
         do {
-          const promises = await this.processOrderBatch(openOrders)
-          const results = await Promise.allSettled(promises)
-          processedOrderError += results.filter((p) => p.status === 'rejected').length
-          totalCheckedOrders += openOrders.orders.length
-        } while (
-          openOrders.cursor &&
-          (openOrders = await this.dbInterface.getByOrderStatus(ORDER_STATUS.OPEN, BATCH_READ_MAX, openOrders.cursor))
-        )
+          asyncLoopCalls.push(this.loopProcess(openOrders))
```
Comment (robert-seitz-uniswap) [inline lib/compute/on-chain-status-checker.ts:47]: parallelized this, need to consider if we get rate limited from infura? @andysmith415

File: lib/handlers/check-order-status/service.ts [hunk 3]
```diff
@@ -64,29 +79,153 @@ export class CheckOrderStatusService {
...
+        ...commonUpdateInfo,
+        ...extraUpdateInfo,
+      }
+      updateList.push(updateObject)
     }
 
-    log.info('validated order', { validation: validation, curBlock: curBlockNumber, orderHash: order.orderHash })
+    updateList.forEach(async (u) => {
+      if (u.orderStatus !== u.lastStatus) {
+        await this.updateStatusAndReturn(u)
```
Comment (robert-seitz-uniswap) [inline lib/handlers/check-order-status/service.ts:178]: add a metric on total time it takes to push a batch to db

File: test/unit/handlers/fill-event-logger.test.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+/* eslint-disable */
```
Comment (robert-seitz-uniswap) [inline test/unit/handlers/fill-event-logger.test.ts:1]: renamed file

File: lib/handlers/check-order-status/fill-event-logger.ts [hunk 5]
```diff
@@ -1,40 +1,35 @@
-import { DutchOrder, FillInfo } from '@uniswap/uniswapx-sdk'
+import { FillInfo } from '@uniswap/uniswapx-sdk'
```
Comment (robert-seitz-uniswap) [inline lib/handlers/check-order-status/fill-event-logger.ts:1]: renamed class

File: bin/stacks/status-stack.ts [hunk 6]
```diff
@@ -36,8 +36,8 @@ export class StatusStack extends cdk.NestedStack {
 
     const taskDefinition = new aws_ecs.FargateTaskDefinition(this, `TaskDef`, {
       taskRole: loaderStackRole,
-      memoryLimitMiB: 2048,
-      cpu: 1024,
+      memoryLimitMiB: 30720,
```
Comment (cgkol) [inline bin/stacks/status-stack.ts:39]: shall we put this back to 8GB?

**Review by cgkol — APPROVED** [review]

File: bin/stacks/status-stack.ts (hunk 6 above)
Comment (robert-seitz-uniswap) [reply bin/stacks/status-stack.ts:39]: yea I can set it back down and profile it

File: lib/compute/on-chain-status-checker.ts [hunk 7]
```diff
@@ -22,44 +27,34 @@ export class OnChainStatusChecker {
...
+        } while (openOrders.cursor && (openOrders = await this.getFromDynamo(openOrders.cursor)))
+
+        await Promise.allSettled(asyncLoopCalls)
```
[the remark this answers is not in the document]
Comment (robert-seitz-uniswap) [reply lib/compute/on-chain-status-checker.ts]: I'll see how 1k looks

File: lib/handlers/check-order-status/service.ts [hunk 8]
```diff
@@ -190,44 +321,16 @@ export class CheckOrderStatusService {
...
-        txHash,
-        settledAmounts,
-        getFillLogAttempts,
-      })
+      log.info('calling updateOrderStatus', { orderHash, orderStatus, lastStatus })
       await this.dbInterface.updateOrderStatus(orderHash, orderStatus, txHash, settledAmounts)
       if (IS_TERMINAL_STATE(orderStatus)) {
         metrics.putMetric(`OrderSfn-${orderStatus}`, 1)
         metrics.putMetric(`OrderSfn-${orderStatus}-chain-${chainId}`, 1)
-        log.info('order in terminal state', {
```
[the remark this answers is not in the document]
Comment (robert-seitz-uniswap) [reply lib/handlers/check-order-status/service.ts:215]: i'll put this one back

File: lib/handlers/check-order-status/service.ts [hunk 9]
```diff
@@ -64,29 +80,155 @@ export class CheckOrderStatusService {
...
+      'OnChainStatusChecker-ValidateBatchQueryTime',
+      MetricUnits.Milliseconds,
+      endTime - startTime
+    )
+
+    const updateList = []
+    startTime = new Date().getTime()
+    const curBlockNumber = await provider.getBlockNumber()
+    const fromBlock = curBlockNumber - FILL_EVENT_LOOKBACK_BLOCKS_ON(chainId)
+    const fillEvents = await orderWatcher.getFillInfo(fromBlock, curBlockNumber)
```
[the remark this answers is not in the document]
Comment (robert-seitz-uniswap) [reply lib/handlers/check-order-status/service.ts:131]: Yea I was thinking this too

File: lib/handlers/check-order-status/service.ts [hunk 10]
```diff
@@ -64,29 +80,155 @@ export class CheckOrderStatusService {
...
+    const updateObject = {
+      ...commonUpdateInfo,
+      ...extraUpdateInfo,
+    }
+
+    return this.updateStatusAndReturn(updateObject)
+  }
+
+  public async batchHandleRequestPerChain(batch: OrderEntity[], chainId: ChainId): Promise<SfnStateInputOutput[]> {
+    const provider = getProvider(chainId)
```
[the remark this answers is not in the document]
Comment (robert-seitz-uniswap) [reply lib/handlers/check-order-status/service.ts:107]: what is that code link?

[the remark this answers is not in the document]
Comment (robert-seitz-uniswap) [reply lib/handlers/check-order-status/service.ts:107]: added a map

**Pull request merged**